primitives/arithmetic/src/fixed.rs

```diff
@@ -0,0 +1,1311 @@
+
+	/// Checked division for integer type `N`. Equal to `self / d`.
+	///
+	/// Returns `None` if the result does not fit in `N` or `d == 0`.
+	fn checked_div_int<N: FixedPointOperand>(self, d: N) -> Option<N>;
+
+	/// Saturating division for integer type `N`. Equal to `self / d`.
+	///
+	/// Panics if `d == 0`. Returns `N::min` or `N::max` if the result does not fit in `N`.
+	fn saturating_div_int<N: FixedPointOperand>(self, d: N) -> N;
+
+	/// Saturating multiplication for integer type `N`, adding the result back.
+	/// Equal to `self * n + n`.
+	///
+	/// Returns `N::min` or `N::max` if the multiplication or final result do not fit in `N`.
+	fn saturating_mul_int_acc<N: FixedPointOperand>(self, n: N) -> N;
+
+	/// Saturating absolute value.
+	///
+	/// Returns `Self::max` if `self == Self::min`.
+	fn saturating_abs(self) -> Self;
+
+	/// Takes the reciprocal (inverse). Equal to `1 / self`.
+	///
+	/// Returns `None` if `self = 0`.
+	fn reciprocal(self) -> Option<Self> {
+		Self::one().checked_div(&self)
+	}
+
+	/// Returns zero.
+	fn zero() -> Self;
+
+	/// Checks if the number is zero.
+	fn is_zero(&self) -> bool;
+
+	/// Returns one.
+	fn one() -> Self;
+
+	/// Checks if the number is positive.
+	fn is_positive(self) -> bool;
+
+	/// Checks if the number is negative.
+	fn is_negative(self) -> bool;
+
+	/// Returns the integer part.
+	fn trunc(self) -> Self;
+
+	/// Returns the fractional part.
+	fn frac(self) -> Self;
+
+	/// Returns the smallest integer greater than or equal to a number.
+	fn ceil(self) -> Self;
+
+	/// Returns the largest integer less than or equal to a number.
+	fn floor(self) -> Self;
+
+	/// Returns the nearest integer to a number. Round half-way cases away from 0.0.
+	fn round(self) -> Self;
+}
+
+struct I129 {
+	value: u128,
+	negative: bool,
+}
+
+impl<N: FixedPointOperand> From<N> for I129 {
+	fn from(n: N) -> I129 {
+		if n < N::zero() {
+			let value: u128 = n.checked_neg()
+				.map(|n| n.unique_saturated_into())
+				.unwrap_or(N::max_value().unique_saturated_into().saturating_add(1));
+			I129 { value, negative: true }
+		} else {
+			I129 { value: n.unique_saturated_into(), negative: false }
+		}
+	}
+}
+
+/// Transforms an `I129` to `N` if it is possible.
+fn from_i129<N: FixedPointOperand>(n: I129) -> Option<N> {
+	let max_plus_one: u128 = N::max_value().unique_saturated_into().saturating_add(1);
+	if n.negative && N::min_value() < N::zero() && n.value == max_plus_one {
+		Some(N::min_value())
+	} else {
+		let unsigned_inner: N = n.value.try_into().ok()?;
+		let inner = if n.negative { unsigned_inner.checked_neg()? } else { unsigned_inner };
+		Some(inner)
+	}
+}
+
+/// Returns `R::max` or `R::min` depending on the operand signs.
+fn to_bound<N: FixedPointOperand, D: FixedPointOperand, R: Bounded>(n: N, m: D) -> R {
+	if (n < N::zero()) != (m < D::zero()) {
+		R::min_value()
+	} else {
+		R::max_value()
+	}
+}
+
+macro_rules! implement_fixed {
+	(
+		$name:ident,
+		$test_mod:ident,
```

Comment:
similar to the other macro for per-things, you can pass a small doc-string here as well. Would be very useful to see the theoretical limit of each type (i.e. Fixed64 can fit at most i64::max_value() / 1000_000_000) here since you refer to it in the other docs often as `if ... exceeds accuracy`.